### ISSUE TITLE
[Console] Native options are not propagated when running a command from another

### ISSUE THREAD

**Issue by welcoMattic** [symfony/symfony#44754]
### Symfony version(s) affected

4.4 and next

### Description

When running a command from another as described in [the documentation](https://symfony.com/doc/current/console/calling_commands.html), the native options passed to the second command are ignored.

Considering 2 commands `app:foo` and `app:bar`.

`app:bar` has 1 option named `--option1`

`app:foo` depends on `app:bar`, and run it with 2 options `--option1` and `--no-interaction`

Option `--option1` of `app:bar` triggers an interaction.

## Scenarios

1. Running `bin/console app:bar` will ask you nothing.

```shell
$ bin/console app:bar


 [OK] app:bar success!


```

2. Running `bin/console app:bar --option1` will ask you

```shell
$ bin/console app:bar --option1

 Are you sure to continue? [yes]:
 >
```

3. Running `bin/console app:bar --option1 --no-interaction` will ask you nothing.

```shell
$ bin/console app:bar --option1 --no-interaction


 [OK] app:bar success!


```

4. Running `bin/console app:foo` or `bin/console app:foo --no-interaction` will always ask you

```shell
$ bin/console app:foo
Do something in app:foo command ...
Run app:bar ...

 Are you sure to continue? [yes]:
 >
```

```shell
$ bin/console app:foo --no-interaction
Do something in app:foo command ...
Run app:bar ...

 Are you sure to continue? [yes]:
 >
```

But the code calling `app:bar` is the following:

```php
$io->writeln('Run app:bar ...');
$this->getApplication()
    ->find('app:bar')
    ->run(new ArrayInput([
        '--option1' => true,
        '--no-interaction' => true,
    ]), $output)
;
```

We explicitly pass `--option1` (to trigger the interactive question) *and* `--no-interaction` to avoid any interaction.
As we saw in the 3rd example, this combination of options should not trigger interaction, but if we call `app:bar` from `app:foo` interaction is always triggered.

### How to reproduce

I made [a reproducer here](https://github.com/welcoMattic/native-options-propagation-reproducer).

### Additional Context

The example is made using `--no-interaction`, it can be by-passed using `$input->setInteractive(false);`. But the issue is still relevant for all other native options, like `--env`, `--quiet`, `--verbose`, `--ansi`, `--no-ansi`, `--no-debug`, ...

**Comment by stof** [comment]
Running a command programmatically from another command is not the intended usage of the symfony/console component, and so there is indeed no support for forwarding the global options, and I don't think we should add such feature.
In most cases, trying to do that indicates that you should extract your logic to a service instead, to be able to use your service in multiple places. 

Btw, `--env` and `--no-debug` won't have any effect, as those are used to boot the kernel, and this forwarding code does not boot a separate kernel.

**Comment by welcoMattic** [comment]
I understand that symfony/console is not intended to support an advanced usage of "command in command". But passing options to a command is kind of basic usage. Maybe can we consider adding this support (except for `--env` and `--no-debug` for reasons you mentioned) as a feature for 6.1?

Use cases can be:

- Database anonymization command which needs to run doctrine migration with `--no-interaction -vvv` 
- Boostraping command that use several maker commands, each one with `quiet` option
- etc 

Basically, I think to achieve this, we need something like [`Application::configureIO`](https://github.com/symfony/symfony/blob/6.1/src/Symfony/Component/Console/Application.php#L888) but for `Command`

**Comment by fabpot** [comment]
What @stof says is that it's not an "advanced usage", but a wrong one. I agree that you'd better extract the logic you want to call in a proper service instead and reuse it. I understand that you mentioned commands from third-parties, but it would be better to refactor those as well to extract their logic.

**Comment by welcoMattic** [comment]
Yes, it is always better to extract logic outside of the commands. The real case that make me writing this issue was from calling `doctrine:migration:migrate` from another command, so I guess I should try to contribute the extraction of the logic in a service to be able to call it from another command instead of calling the command itself.

Thanks for the discussion :+1:, I close this issue.
